### ISSUE TITLE
Wrong bbox (axis order) generated with GDAL 3 causes empty tilesets

### ISSUE THREAD

**Issue by kannes** [t-rex-tileserver/t-rex#205]
For some reason t-rex generate does not store generated tiles from zoom levels higher than 2 for me. It has no trouble generating them, I see the progress bars filling up and via `serve` everything is fine too.

It looks to me like it might drop tiles because they are empty for some reason? I noticed that the generated metadata.json contains the bounds in WGS84 even if the SRS is something else like EPSG:3857. Could this cause trouble? I also noticed that the demo pages available in the web browser page it loads when using `serve` do not manage to load any tiles.

The feature count on the layer is about 30,000 polygons, with many geometric duplicates.

I could share the data in private, ~5MB.

t-rex is 2be4a6edc690f28e675124504470bfc3fb3becba built last weekend.

# t-rex-config.toml
```
# t-rex configuration

[service.mvt]
viewer = true

[[datasource]]
path = "../my_data.gpkg"
name = "datasource"
default = true

[grid]
predefined = "web_mercator"

[[tileset]]
name = "my_data"
extent = [53.40387, 8.49394, 53.91590, 10.31846]
[[tileset.layer]]
name = "my_data"
table_name = "my_data"
geometry_field = "geom"
geometry_type = "POLYGON"
srid = 25832
buffer_size = 1
#make_valid = true
simplify = true
#query_limit = 1000
#[[tileset.layer.query]]

[cache.file]
base = "/tmp/mvtcache"
#baseurl = "http://example.com/tiles"

[webserver]
# Bind address. Use 0.0.0.0 to listen on all adresses.
bind = "127.0.0.1"
port = 6767

[[webserver.static]]
path = "/www"
dir = "./www/"
```

# Generator invocation for zoom level 3
```
$ cargo run -- generate --config ../t-rex-config.toml --maxzoom 3 --minzoom 3
    Finished dev [unoptimized + debuginfo] target(s) in 0.15s
     Running `target/debug/t_rex generate --config ../t-rex-config.toml --maxzoom 3 --minzoom 3`
2020-08-24 22:06:18.827 INFO Reading configuration from '../t-rex-config.toml'
2020-08-24 22:06:18.844 INFO Layer 'my_data': Reprojecting geometry to SRID 3857
2020-08-24 22:06:18.866 WARN Layer 'my_data': Simplification not supported for GDAL layers
2020-08-24 22:06:18.866 WARN Layer 'my_data': Clipping with buffer_size not supported for GDAL layers
2020-08-24 22:06:18.867 INFO Tile cache directory: /tmp/mvtcache
Generating tileset 'my_data'...
Level 3: 1 / 1 [=============================================================]  
Statistics:
feature_count.my_data.my_data.3: #measurements: 1, min: 0, max: 0, mean: 0.0000000000 +/- 0.0000000000
tile_ms.my_data.my_data.3: #measurements: 1, min: 44, max: 44, mean: 44.0000000000 +/- 0.0000000000
```

# Result
```
$ tree /tmp/mvtcache
/tmp/mvtcache
├── my_data
│   └── metadata.json
├── my_data.json
└── my_data.style.json

1 directory, 3 files
```

# Zoomlevel 0 works fine
```
$ cargo run -- generate --config ../t-rex-config.toml --maxzoom 0 --minzoom 0
    Finished dev [unoptimized + debuginfo] target(s) in 0.16s
     Running `target/debug/t_rex generate --config ../t-rex-config.toml --maxzoom 0 --minzoom 0`
2020-08-24 22:08:46.868 INFO Reading configuration from '../t-rex-config.toml'
2020-08-24 22:08:46.885 INFO Layer 'my_data': Reprojecting geometry to SRID 3857
2020-08-24 22:08:46.907 WARN Layer 'my_data': Simplification not supported for GDAL layers
2020-08-24 22:08:46.907 WARN Layer 'my_data': Clipping with buffer_size not supported for GDAL layers
2020-08-24 22:08:46.908 INFO Tile cache directory: /tmp/mvtcache
Generating tileset 'my_data'...
Level 0: 1 / 1 [=============================================================]  
Statistics:
feature_count.my_data.my_data.0: #measurements: 1, min: 5939, max: 5939, mean: 5939.0000000000 +/- 0.0000000000
tile_ms.my_data.my_data.0: #measurements: 1, min: 44252, max: 44252, mean: 44252.0000000000 +/- 0.0000000000
```

```
$ tree /tmp/mvtcache
/tmp/mvtcache
├── my_data
│   ├── 0
│   │   └── 0
│   │       └── 0.pbf
│   └── metadata.json
├── my_data.json
└── my_data.style.json

3 directories, 4 files
```

# Zoom level 2 as well
```
$ cargo run -- generate --config ../t-rex-config.toml --maxzoom 2 --minzoom 2
    Finished dev [unoptimized + debuginfo] target(s) in 0.15s
     Running `target/debug/t_rex generate --config ../t-rex-config.toml --maxzoom 2 --minzoom 2`
2020-08-24 22:11:32.877 INFO Reading configuration from '../t-rex-config.toml'
2020-08-24 22:11:32.894 INFO Layer 'my_data': Reprojecting geometry to SRID 3857
2020-08-24 22:11:32.916 WARN Layer 'my_data': Simplification not supported for GDAL layers
2020-08-24 22:11:32.916 WARN Layer 'my_data': Clipping with buffer_size not supported for GDAL layers
2020-08-24 22:11:32.916 INFO Tile cache directory: /tmp/mvtcache
Generating tileset 'my_data'...
Level 2: 1 / 1 [=============================================================]  
Statistics:
feature_count.my_data.my_data.2: #measurements: 1, min: 29340, max: 29340, mean: 29340.0000000000 +/- 0.0000000000
tile_ms.my_data.my_data.2: #measurements: 1, min: 991720, max: 991720, mean: 991720.0000000000 +/- 0.0000000000
```

```
$ tree /tmp/mvtcache
/tmp/mvtcache
├── my_data
│   ├── 2
│   │   └── 2
│   │       └── 1.pbf
│   └── metadata.json
├── my_data.json
└── my_data.style.json

3 directories, 4 files
```

**Comment by pka** [comment]
Hi Johannes,

Can you send me your data to my Mail adress (see GH profile)?
We could also do a support session, if you're interested in.

Pirmin

**Comment by kannes** [comment]
Sent!

A support session would be a waste of your time ;) It's just a little toy project I am toying with.

**Comment by pka** [comment]
I was able to reproduce your problem:

```
cargo run -- generate --config config.toml --minzoom 2 --maxzoom 3 --loglevel debug

2020-09-03 22:24:08.920 INFO Reading configuration from 'config.toml'
2020-09-03 22:24:08.934 INFO Layer 'my_data': Reprojecting geometry to SRID 3857
2020-09-03 22:24:08.951 WARN Layer 'my_data': Simplification not supported for GDAL layers
2020-09-03 22:24:08.951 WARN Layer 'my_data': Clipping with buffer_size not supported for GDAL layers
2020-09-03 22:24:08.952 INFO Tile cache directory: /tmp/mvtcache
2020-09-03 22:24:08.952 DEBUG Filecache.write /tmp/mvtcache/my_data.json
2020-09-03 22:24:08.952 DEBUG Filecache.write /tmp/mvtcache/my_data.style.json
2020-09-03 22:24:08.953 DEBUG Filecache.write /tmp/mvtcache/my_data/metadata.json
Generating tileset 'my_data'...
2020-09-03 22:24:08.953 DEBUG wgs84 extent: Some(Extent { minx: 53.40387, miny: 8.49394, maxx: 53.9159, maxy: 10.31846 })
2020-09-03 22:24:08.953 DEBUG tile limits: Extent { minx: 5944891.614790178, miny: 949023.6238184517, maxx: 6001890.533661058, maxy: 1154905.5026989975 }
2020-09-03 22:24:08.953 DEBUG level 2: ExtentInt { minx: 2, miny: 2, maxx: 3, maxy: 3 }
Level 2: 0 / 1 [---------------------------------------------------------------------------------------------------------------------------------------------------------]  2020-09-03 22:24:08.953 DEBUG my_data/2/2/2 retrieving with Extent { minx: 0.0, miny: 0.0, maxx: 10018754.171394624, maxy: 10018754.171394624 }
2020-09-03 22:24:08.955 DEBUG retrieve_features layer: baugenehmigungen
2020-09-03 22:39:01.593 DEBUG my_data/2/2/2 layer my_data: 29340 features
2020-09-03 22:39:05.953 DEBUG Filecache.write /tmp/mvtcache/my_data/2/2/1.pbf
Level 2: 1 / 1 [=========================================================================================================================================================]  2020-09-03 22:39:05.967 DEBUG level 3: ExtentInt { minx: 5, miny: 4, maxx: 6, maxy: 5 }
Level 3: 0 / 1 [---------------------------------------------------------------------------------------------------------------------------------------------------------]  2020-09-03 22:39:05.967 DEBUG my_data/3/5/4 retrieving with Extent { minx: 5009377.085697312, miny: 0.0, maxx: 10018754.171394624, maxy: 5009377.085697312 }
2020-09-03 22:39:05.968 DEBUG retrieve_features layer: baugenehmigungen
2020-09-03 22:39:05.993 DEBUG my_data/3/5/4 layer my_data: 0 features
Level 3: 1 / 1 [=========================================================================================================================================================]  
Statistics:
feature_count.my_data.my_data.2: #measurements: 1, min: 29340, max: 29340, mean: 29340.0000000000 +/- 0.0000000000
feature_count.my_data.my_data.3: #measurements: 1, min: 0, max: 0, mean: 0.0000000000 +/- 0.0000000000
tile_ms.my_data.my_data.2: #measurements: 1, min: 892639, max: 892639, mean: 892639.0000000000 +/- 0.0000000000
tile_ms.my_data.my_data.3: #measurements: 1, min: 25, max: 25, mean: 25.0000000000 +/- 0.0000000000
```

So t-rex doesn't find a feature in tile 3/5/4 { minx: 5009377.085697312, miny: 0.0, maxx: 10018754.171394624, maxy: 5009377.085697312 }.

One guess is, that the changed axis order in GDAL 3 causes a wrong bbox transformation result. Will investigate this.

**Comment by pka** [comment]
My guess was wrong. Changing the tileset extent to `[8.49394, 53.40387, 10.31846, 53.91590]` does the trick!

**Comment by pka** [comment]
Seems like `t-rex genconfig` generates a wrong extent with GDAL 3. So it is a bug

**Comment by kannes** [comment]
Glad to hear that it wasn't error on my end. Thanks for investigating. :) 
That axis order change sure ripples a long way through all the tools.

I will edit my config and see what happens.

**Comment by pka** [comment]
Fixed in 8609c09
